deployments/helm-chart/templates/controller-daemonset.yaml

```diff
@@ -238,8 +252,39 @@ spec:
 {{- if .Values.controller.extraContainers }}
       {{ toYaml .Values.controller.extraContainers | nindent 6 }}
 {{- end }}
+{{- if .Values.controller.readOnlyRootFilesystem }}
+      volumes:
+      - name: nginx-etc
+        emptyDir: {}
+      - name: nginx-cache
+        emptyDir: {}
+      - name: nginx-lib
+        emptyDir: {}
+      - name: nginx-log
+        emptyDir: {}
+{{- end }}
```

Comment:
Helm is giving errors when both `controller.readOnlyRootFilesystem: true` and `nginxServiceMesh.enable: true` are set.
On L66 we add a `volumes` block if we set either `controller.volumes` or `nginxServiceMesh.enable`

https://github.com/nginxinc/kubernetes-ingress/pull/3548/files#diff-2eccfbc1886b3803b5035aad15dc3f9f38ecfd0b5c5a540408c828f32d6ab3d3R66-R74

This ends up conflicting with the `volumes` block here on L256
We can fix this by removing the `volumes` block on L256 and then moving the rest of this block below L74 in the mentioned link.

We can also add `.Values.controller.readOnlyRootFilesystem` to the condition block that adds the `volumes` block.

So all together it should look like this:

```
{{- if or .Values.controller.volumes .Values.nginxServiceMesh.enable .Values.controller.readOnlyRootFilesystem}}
      volumes:
{{- end }}
{{- if .Values.nginxServiceMesh.enable }}
      - hostPath:
          path: /run/spire/sockets
          type: DirectoryOrCreate
        name: spire-agent-socket
{{- end }}
{{- if .Values.controller.readOnlyRootFilesystem }}
      volumes:
      - name: nginx-etc
        emptyDir: {}
      - name: nginx-cache
        emptyDir: {}
      - name: nginx-lib
        emptyDir: {}
      - name: nginx-log
        emptyDir: {}
{{- end }}
```

Reply:
Same change can be applied to `controller-deployment.yaml`

Reply:
Yeah. I applied this to the `volumeMounts` but missed it for the `volumes` themselves.

Reply:
Latest commit on the PR resolves this. Thank you!